lib/fshash/bucket.go

```diff
@@ -58,9 +15,30 @@ func (m Metadata) MarshalBinary() ([]byte, error) {
 */
 type Bucket interface {
 	Record(metadata Metadata, contentHash []byte)
+	Iterator() (rootRecord RecordIterator)
```

Comment:
Some one-liner docstrings here would be nice, in particular clarifying (unless I misunderstand) that Iterator walks an assembled bucket for purposes of hashing after assembly.

In fact, nothing here seems to deal with the assembly of a bucket, that's punted to implementations, yes?